src/main/java/org/opensearch/ad/task/ADBatchTaskRunner.java

```diff
@@ -968,9 +968,10 @@ private void detectAnomaly(
                 String error = feature.getUnprocessedFeatures().isPresent()
                     ? "No full shingle in current detection window"
                     : "No data in current detection window";
+                String resultTaskId = adTask.getParentTaskId() != null ? adTask.getParentTaskId() : adTask.getTaskId();
```

Comment:
If a task has no parent task, it is the detector level task, right?

Reply:
Yes